app/services/extension.js

```diff
@@ -57,10 +63,48 @@ export default Service.extend({
 
   setup(extension) {
     set(this, 'data', extension.field.getValue() || {});
-    console.log(
-      extension.parameters.instance.schemaShorthand, 'shorthand'
-    );
     set(this, 'extension', extension);
+    this.loadSchemaFromShorthand();
+    this.syncFragmentsToSchema();
+  },
+
+  setSetting(key, value) {
+    set(this, 'data._settings', get(this, 'data._settings') || {});
+    set(this, `data._settings.${key}`, value);
+  },
+
+  loadSchemaFromShorthand() {
+    const shorthand = (get(this, 'extension.parameters.instance.schemaShorthand') || "");
+    if (!shorthand.length) return;
+
+    const parsedSchemaFields = 
+      shorthand.split(",").map(tuple => tuple.split(":").map(t => t.trim()));
+    const existingSchema = get(this, 'data._schema') || [];
+
+    const loadedSchema = parsedSchemaFields.reduce((acc, fieldTuple) => {
+      const [key, type] = fieldTuple;
+      if (key.length === 0) {
+        console.warn(
```

Comment:
these are throwing errors in the tests... if you want to keep them, mb add an eslint-ignore here, ando n line 93?